application/src/Serde.tsx

```diff
@@ -1,480 +1,494 @@
-// TODO renmare serialied(string,num) to serializedData (since other thing may be present)
+function rangeArr(rangeStart: number, rangeEnd: number): number[] {
+  return Array(rangeEnd - rangeStart)
+    .fill(0)
+    .map((_, i) => i + rangeStart);
+}
 
-function combineByteArrays(array1: Uint8Array, array2: Uint8Array): Uint8Array {
-  let combined: Uint8Array = new Uint8Array(array1.length + array2.length);
+class BitArray {
+  boolArr: Array<boolean>;
+  bitCount: number;
 
-  let i = 0;
-  array1.forEach((byte) => (combined[i++] = byte));
-  array2.forEach((byte) => (combined[i++] = byte));
+  constructor(arr?: Uint8Array) {
+    this.bitCount = 0;
+    this.boolArr = new Array();
+    if (arr !== undefined) {
+      this.insert(arr, arr.length * 8);
+    }
+  }
 
-  return combined;
-}
+  insert(data: Uint8Array, totalBitCount: number): void;
+  insert(bitArray: BitArray): void;
+  insert(data: BitArray | Uint8Array, totalBitCount?: number) {
```

Comment:
Maybe make another overload that gets multiple bit arrays and inserts them

Reply:
Why is this needed?